walk/walk.go

```diff
@@ -186,12 +183,13 @@ func visit(c *config.Config, cexts []config.Configurer, knownDirectives map[stri
 	shouldUpdate := updateRels.shouldUpdate(rel, updateParent)
 	for _, sub := range subdirs {
 		if subRel := path.Join(rel, sub); updateRels.shouldVisit(subRel, shouldUpdate) {
-			visit(c, cexts, knownDirectives, updateRels, trie.children[sub], wf, subRel, shouldUpdate)
+			visit(c.Clone(), cexts, knownDirectives, updateRels, trie.children[sub], wf, subRel, shouldUpdate)
```

Comment:
It looks like we cloned only once (in `configure`) before this change, but now clone once for every subdir. Was that a bug (we should have cloned before) or why is this not a potential performance concern?

Reply:
Each `visit` is essentially just a `configure` though. However now I wonder (and forget) if this will conflict with https://github.com/bazel-contrib/bazel-gazelle/pull/1921 so I think we should focus on that PR first.